Code/Framework/AzFramework/AzFramework/Physics/HeightfieldProviderBus.h

```diff
@@ -115,9 +118,19 @@ namespace Physics
     public:
         static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Multiple;
 
+        enum class HeightfieldChangeMask
+        {
+            None = 0b00000000,
+            Settings = 0b00000001,
+            HeightData = 0b00000010,
+            MaterialData = 0b00000100,
+            SurfaceData = 0b00001000
+        };
+
         //! Called whenever the heightfield data changes.
         //! @param the AABB of the area of data that changed.
-        virtual void OnHeightfieldDataChanged([[maybe_unused]] const AZ::Aabb& dirtyRegion)
+        virtual void OnHeightfieldDataChanged([[maybe_unused]] const AZ::Aabb& dirtyRegion, 
+            [[maybe_unused]] const Physics::HeightfieldProviderNotifications::HeightfieldChangeMask* changeMask)
```

Comment:
If the only reason to make this a pointer is for some backwards compatibility, I'd encourage you to instead just make it a value. We're still actively changing the terrain system, so I don't think we need to be overly concerned with API backwards compatibility yet.